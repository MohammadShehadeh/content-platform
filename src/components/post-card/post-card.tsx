import { ArrowRight } from 'lucide-react';
import Link from 'next/link';

import { Button } from '@/components/shared/button';
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from '@/components/shared/card';
import type { Post } from '@/services/posts';

import styles from './post-card.module.scss';

type PostCardProps = Pick<Post, 'title' | 'body' | 'id'> & {
  className?: string;
};

export const PostCard = ({ title, body, id, className }: PostCardProps) => {
  return (
    <Card className={className}>
      <CardHeader>
        <CardTitle>{title}</CardTitle>
        <CardDescription className={styles.description}>{body}</CardDescription>
      </CardHeader>
      <CardContent>
        <Button variant="link" size="sm" asChild>
          <Link href={`/posts/${id}`} className={styles.link}>
            View Post
            <ArrowRight width={16} height={16} />
          </Link>
        </Button>
      </CardContent>
    </Card>
  );
};
