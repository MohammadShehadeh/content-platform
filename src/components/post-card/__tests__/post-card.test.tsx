import { render, screen } from '@testing-library/react';

import { PostCard } from '@/components/post-card';

describe('PostCard', () => {
  it('should render with all props correctly', () => {
    render(<PostCard title="Test Title" body="Test Body" id={42} className="custom-class" />);

    expect(screen.getByText('Test Title')).toBeInTheDocument();
    expect(screen.getByText('Test Body')).toBeInTheDocument();
    expect(screen.getByText('View Post')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/posts/42');
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard title="Hello" body="Hello" id={1} />);
    expect(container).toMatchSnapshot();
  });
});
