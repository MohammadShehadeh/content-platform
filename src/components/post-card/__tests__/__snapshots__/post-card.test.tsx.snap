// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard matches snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        Hello
      </h3>
      <p
        class="description description"
      >
        Hello
      </p>
    </div>
    <div
      class="content"
    >
      <a
        class="button link sm link"
        href="/posts/1"
      >
        View Post
        <svg
          class="lucide lucide-arrow-right"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="m12 5 7 7-7 7"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`PostCard should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        Hello
      </h3>
      <p
        class="description description"
      >
        Hello
      </p>
    </div>
    <div
      class="content"
    >
      <a
        class="button link sm link"
        href="/posts/1"
      >
        View Post
        <svg
          class="lucide lucide-arrow-right"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="m12 5 7 7-7 7"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
